### ISSUE TITLE
Revision age

### ISSUE THREAD

**Issue by cstjean** [timholy/Revise.jl#377]
I use the world age to invalidate my caches quite a bit (as in: I put the world-age in the key to force a recomputation whenever the code has changed). It works nicely, but it can be overly conservative (eg. `julia> map(x->x+1, 1:3)` changes the world age). If there was a `revision_age` counter, that gets increased at each revision, I would make some use of it in my code. And it shouldn't be hard to do, right? I'd be happy to contribute it.

**Comment by cstjean** [comment]
Alternatively, is there already some way of knowing when the last revised code was?

**Comment by timholy** [comment]
There's the [logging framework](https://timholy.github.io/Revise.jl/stable/debugging/), which might be coerced into doing something along your lines. I'm not sure I fully understand the question, so yes, it might be necessary for you to illustrate with code.

**Comment by cstjean** [comment]
In functional code,

```julia
memo = Dict()
foo(x) = get!(memo, x) do
    ... call `bar`
end
```

is a nice way of memoizing a function's result, but if the `bar` definition changes, the cache should be invalidated. An easy, conservative way to ensure that is to use `get!(memo, (x, ccall(:jl_get_world_counter, UInt, ())) do ...`  When working interactively, the world age gets incremented a lot (anytime a closure is created), even though it cannot affect the result of `foo(1)`. I'd like to replace `get_world_counter` with `Revise.get_revision_counter()` so that the cache is only invalidated when code has been changed.

I've realized that I could probably use `entr`. Is there a way to follow all modules/files?
